src/guides/v2.3/extension-dev-guide/message-queues/async-topics.md

```diff
@@ -0,0 +1,68 @@
+---
+group: php-developer-guide
+title: Topics in Asynchronous API
+contributor_name: comwrap GmbH
+contributor_link: https://www.comwrap.com
+functional_areas:
+  - Services
+---
+
+Magento Queuing system is using topics exchange for managing messages. More detailed information about topics can be found [here](https://www.rabbitmq.com/tutorials/tutorial-five-python.html){:target="_blank"}.
+
+In Magento, topics are usually defined in communication.xml configuration files. See [Configure message queues]({{ page.baseurl }}/guides/v2.4/extension-dev-guide/message-queues/config-mq.html#communicationxml) https://devdocs.magento.com/guides/v2.4/extension-dev-guide/message-queues/config-mq.html#communicationxml
+
+### Generating of communication.xml
+
+`communication.xml` for asynchronous api, is pre-generated automatically by `WebapiAsync` module. This module also automatically generating topic names for asynchronous processes. File generation processed by `\Magento\WebapiAsync\Code\Generator\Config\RemoteServiceReader\Communication` class, which implements `\Magento\Framework\Config\ReaderInterface` and injected into `\Magento\Framework\Communication\Config\CompositeReader` constructor argument of main `di.xml` file.
+
+```xml
+<type name="Magento\Framework\Communication\Config\CompositeReader">
+    <arguments>
+        <argument name="readers" xsi:type="array">
+            <item name="asyncServiceReader" xsi:type="array">
+                <item name="reader" xsi:type="object">Magento\WebapiAsync\Code\Generator\Config\RemoteServiceReader\Communication</item>
+                <item name="sortOrder" xsi:type="string">0</item>
+            </item>
+            <item name="xmlReader" xsi:type="array"> ... </item>
+            ...
+        </argument>
+    </arguments>
+</type>
+```
+
+Sort order is set to 0 and this allow developers to change some aspect of generated configuration in following config readers such as `communication.xml`, `env.php`.
+
+`\Magento\Framework\Communication\Config\CompositeReader::read()` collects configs from defined readers and merge config from them to one config.
+
+Because generation of topics config is based on schema type, generated `<topic>` xml, by default will be returned with `"sync"=true` and `response` based on service response definition. So `WebapiAsync` module will change those settings to `"sync"=false` and `response` set to null. This changes will allow to execute topics in asynchronous mode.
```

Comment:
```suggestion
Because the generation of topics config is based on schema type, generated `<topic>` XML, by default will be returned with `"sync"=true` and `response` based on service response definition. So `WebapiAsync` module will change those settings to `"sync"=false` and `response` set to null. These changes will allow to execute topics in asynchronous mode.
```